RoboJackets/igvc-software
# Feat/pointcloud clustering

# Description
This project contains a software package that segments lidar point cloud into clusters and creates a bounding box over each cluster in real time. This package adds a feature to localize objects surrounded by the robot such as barrels and people. Here is [readme file](https://github.com/RoboJackets/igvc-software/tree/feat/pointcloud_clustering/igvc_perception/src/pointcloud_segmentation) for this project.

This PR does the following:
- Ground plane segmentation
- Point cloud clustering that categorizes lidar points into several clusters using two well known method (region growing and euclidean clustering methods)
- Visualization tool that creates a bounding box over each clusters

Fixes  #777 and #651 

# Testing steps (If relevant)
## Test Case 1
1. Download a bag file (e.g. JESSII_RUN_SUN_2_2019-06-10-15-30-13.bag) from the robojacket cloud
2. Modify launch file and add your file path
3. Run the following command to visualize the point cloud segmentation.
`roslaunch igvc_perception pointcloud_filter.launch` 
`roslaunch igvc_perception ptseg.launch` 


Expectation: You should see the following result in your rviz screen. Each cluster has its own bounding box.

![image](https://user-images.githubusercontent.com/35343680/102704314-caf38400-4247-11eb-9914-3d48a144c613.png)

Note: here is [readme file](https://github.com/RoboJackets/igvc-software/tree/feat/pointcloud_clustering/igvc_perception/src/pointcloud_segmentation) for this project.

# Self Checklist
